File: JavaSource/src/main/java/ru/akhcheck/patterns/visitor/workers/Painter.java
package ru.akhcheck.patterns.visitor.workers;

import ru.akhcheck.patterns.visitor.rooms.BathRoom;
import ru.akhcheck.patterns.visitor.rooms.Home;
import ru.akhcheck.patterns.visitor.rooms.Kitchen;
import ru.akhcheck.patterns.visitor.rooms.LivingRoom;

/** Маляр: красит стены и потолки, пол не трогает. */
public class Painter implements Visitor {
    @Override
    public void visit(Kitchen kitchen) {
        kitchen.getCeiling().setColor("white");
    }

    @Override
    public void visit(LivingRoom livingRoom) {
        livingRoom.getWall().setColor("beige");
        livingRoom.getCeiling().setColor("white");
    }

    @Override
    public void visit(BathRoom bathRoom) {
        bathRoom.getCeiling().setColor("white");
    }

    @Override
    public void visit(Home home) {
        System.out.println("Painter came to the house");
    }
}

File: Visitor/python-source/workers/Painter.py
from __future__ import annotations

from typing import TYPE_CHECKING

from .Visitor import Visitor

if TYPE_CHECKING:
    from rooms import BathRoom
    from rooms import Home
    from rooms import Kitchen
    from rooms import LivingRoom


class Painter(Visitor):
    """Маляр: красит стены и потолки, пол не трогает."""

    def visit_kitchen(self, kitchen: Kitchen) -> None:
        kitchen.ceiling.color = "white"

    def visit_living_room(self, living_room: LivingRoom) -> None:
        living_room.wall.color = "beige"
        living_room.ceiling.color = "white"

    def visit_bath_room(self, bath_room: BathRoom) -> None:
        bath_room.ceiling.color = "white"

    def visit_home(self, home: Home) -> None:
        print("Painter came to the house")
